cfme/tests/ssui/test_ssui_dashboard.py

```diff
@@ -27,6 +32,108 @@
+            'Used Disk I/O': {'per_time': 'Hourly', 'variable_rate': '2'},
+            'Allocated Memory': {'per_time': 'Hourly', 'fixed_rate': '1'},
+            'Used Memory': {'per_time': 'Hourly', 'variable_rate': '2'}})
+        compute.create()
+        storage = rates.StorageRate(description=desc, fields={
+            'Used Disk Storage': {'per_time': 'Hourly', 'variable_rate': '3'},
+            'Allocated Disk Storage': {'per_time': 'Hourly', 'fixed_rate': '3'}})
+        storage.create()
+        yield desc
+
+    finally:
+        compute.delete()
+        storage.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def assign_chargeback_rate(new_compute_rate):
+    # Assign custom Compute rate to the Enterprise and then queue the Chargeback report.
+    description = new_compute_rate
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            'Enterprise': {'Rate': description}
+        })
+    enterprise.computeassign()
+    enterprise.storageassign()
+    logger.info('Assigning CUSTOM Compute and Storage rates')
+
+    yield
+
+    # Resetting the Chargeback rate assignment
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            'Enterprise': {'Rate': '<Nothing>'}
+        })
+    enterprise.computeassign()
+    enterprise.storageassign()
+
+
+@pytest.fixture(scope="function")
+def run_service_chargeback_report(provider, appliance, assign_chargeback_rate,
+        order_catalog_item_in_ops_ui):
+    catalog_item = order_catalog_item_in_ops_ui
+    vmname = '{}0001'.format(catalog_item.provisioning_data['catalog']["vm_name"])
+
+    def verify_records_rollups_table(appliance, provider):
+        # Verify that hourly rollups are present in the metric_rollups table
+        # before running Service Chargeback report.
+
+        ems = appliance.db.client['ext_management_systems']
+        rollups = appliance.db.client['metric_rollups']
+        with appliance.db.client.transaction:
+            result = (
+                appliance.db.client.session.query(rollups.id)
+                .join(ems, rollups.parent_ems_id == ems.id)
+                .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vmname,
+                ems.name == provider.name, rollups.timestamp >= date.today())
+            )
+
+        for record in appliance.db.client.session.query(rollups).filter(
+                rollups.id.in_(result.subquery())):
+            if (record.cpu_usagemhz_rate_average is not None or
+               record.cpu_usage_rate_average is not None or
+               record.derived_memory_used is not None or
+               record.net_usage_rate_average is not None or
+               record.disk_usage_rate_average):
+                return True
+
+        return False
+
+    wait_for(verify_records_rollups_table, [appliance, provider], timeout=3600,
+        fail_condition=False, message='Waiting for hourly rollups')
```

Comment:
**REQUIRED**
`fail_condition=False` is not needed

Reply:
@quarckster , I've addressed all your comments.